runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -5562,6 +5562,10 @@ static void genHeapAlloc(TR::Node *node, TR::Instruction *&iCursor, TR_OpaqueCla
 
                iCursor = generateTrg1MemInstruction(cg,TR::InstOpCode::Op_load, node, heapTopReg,
                      TR::MemoryReference::createWithDisplacement(cg, metaReg, offsetof(J9VMThread, heapTop), TR::Compiler->om.sizeofReferenceAddress()), iCursor);
+
+               if (needZeroInit)
+                   iCursor = generateTrg1ImmInstruction(cg, TR::InstOpCode::li, node, zeroReg, 0, iCursor);
```

Comment:
From the tracelogs, r2 was used for len and not r7. So will set needZeroInit to false in case of zeroLen array initialization as r7 will not be used for data initialization.

Reply:
```
0x78a9b1768100 {TestAllocArrayAfterAllocNoUse.test} +0                        50006fe9 ld        r11, 0x50(r15) J9VMThread.stackOverflowMark // stack or async check                      0x78a9b1768104 {TestAllocArrayAfterAllocNoUse.test} +1                        e0ffce39 addi      r14, r14, -0x20    
0x78a9b1768108 {TestAllocArrayAfterAllocNoUse.test} +2                        a602087c mflr      r0                                            
0x78a9b176810c {TestAllocArrayAfterAllocNoUse.test} +3                        40582e7c cmpld     r14, r11                                      
0x78a9b1768110 {TestAllocArrayAfterAllocNoUse.test} +4                        18000ef8 std       r0, 0x18(r14)                       
0x78a9b1768114 {TestAllocArrayAfterAllocNoUse.test} +5                        3401c140 ble-      0x78a9b1768248 C>> +82                            
0x78a9b1768118 {TestAllocArrayAfterAllocNoUse.test} +6                        600770e8 ld        r3, 0x760(r16) JITTOC 0x78a9c80d63a8 = 0x0  <<< ^+86                                     
0x78a9b176811c {TestAllocArrayAfterAllocNoUse.test} +7                        00006380 lwz       r3, 0(r3)                                                                                0x78a9b1768120 {TestAllocArrayAfterAllocNoUse.test} +8                        0000032c cmpwi     r3, 0                                  
0x78a9b1768124 {TestAllocArrayAfterAllocNoUse.test} +9                   *    b4008240 bne       0x78a9b17681d8 C>> +54                        
0x78a9b1768128 {TestAllocArrayAfterAllocNoUse.test} +10                  |>   00004038 li        r2, 0 <<< ^+62 ^+74               
0x78a9b176812c {TestAllocArrayAfterAllocNoUse.test} +11                  ||   60000fe9 ld        r8, 0x60(r15) J9VMThread.heapAlloc // TLH Alloc                                          
0x78a9b1768130 {TestAllocArrayAfterAllocNoUse.test} +12                  ||   68008fe8 ld        r4, 0x68(r15) J9VMThread.heapTop                                                         0x78a9b1768134 {TestAllocArrayAfterAllocNoUse.test} +13                  ||   18000838 addi      r0, r8, 0x18             
0x78a9b1768138 {TestAllocArrayAfterAllocNoUse.test} +14                  ||   4020207c cmpld     r0, r4                            
0x78a9b176813c {TestAllocArrayAfterAllocNoUse.test} +15                  ||   f0008141 bgt       0x78a9b176822c C>> +75            
0x78a9b1768140 {TestAllocArrayAfterAllocNoUse.test} +16                  ||   60000ff8 std       r0, 0x60(r15) J9VMThread.heapAlloc
0x78a9b1768144 {TestAllocArrayAfterAllocNoUse.test} +17                  ||   780890e8 ld        r4, 0x878(r16) JITTOC 0xbaa00 Ptr 
0x78a9b1768148 {TestAllocArrayAfterAllocNoUse.test} +18                  ||   00008890 stw       r4, 0(r8)                      
0x78a9b176814c {TestAllocArrayAfterAllocNoUse.test} +19                  ||   04004890 stw       r2, 4(r8)                                                                                0x78a9b1768150 {TestAllocArrayAfterAllocNoUse.test} +20                  ||   08004890 stw       r2, 8(r8)                                                                                0x78a9b1768154 {TestAllocArrayAfterAllocNoUse.test} +21                  ||   18006839 addi      r11, r8, 0x18
0x78a9b1768158 {TestAllocArrayAfterAllocNoUse.test} +22                  ||   100068f9 std       r11, 0x10(r8)                                                                            0x78a9b176815c {TestAllocArrayAfterAllocNoUse.test} +23                  ||>  00000461 ori       r4, r8, 0 <<< ^+81                                                                       0x78a9b1768160 {TestAllocArrayAfterAllocNoUse.test} +24                  |||  ac04207c lwsync     
0x78a9b1768164 {TestAllocArrayAfterAllocNoUse.test} +25                  |||  0100703c addis     r3, r16, 1 JITTOC 0x4ea700 Ptr      
```